components/api/src/test/java/com/hotels/styx/api/messages/FullHttpResponseTest.java

```diff
@@ -0,0 +1,405 @@
+import io.netty.handler.codec.http.HttpResponseStatus;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+import rx.observers.TestSubscriber;
+
+import java.util.Optional;
+
+import static com.google.common.net.MediaType.ANY_AUDIO_TYPE;
+import static com.hotels.styx.api.HttpCookie.cookie;
+import static com.hotels.styx.api.HttpCookieAttribute.domain;
+import static com.hotels.styx.api.HttpCookieAttribute.maxAge;
+import static com.hotels.styx.api.HttpCookieAttribute.path;
+import static com.hotels.styx.api.HttpHeader.header;
+import static com.hotels.styx.api.HttpHeaderNames.CONTENT_LENGTH;
+import static com.hotels.styx.api.HttpHeaderNames.LOCATION;
+import static com.hotels.styx.api.matchers.HttpHeadersMatcher.isNotCacheable;
+import static com.hotels.styx.support.matchers.IsOptional.isAbsent;
+import static com.hotels.styx.support.matchers.IsOptional.isValue;
+import static io.netty.buffer.Unpooled.copiedBuffer;
+import static io.netty.handler.codec.http.HttpResponseStatus.BAD_GATEWAY;
+import static io.netty.handler.codec.http.HttpResponseStatus.BAD_REQUEST;
+import static io.netty.handler.codec.http.HttpResponseStatus.CREATED;
+import static io.netty.handler.codec.http.HttpResponseStatus.GATEWAY_TIMEOUT;
+import static io.netty.handler.codec.http.HttpResponseStatus.MOVED_PERMANENTLY;
+import static io.netty.handler.codec.http.HttpResponseStatus.MULTIPLE_CHOICES;
+import static io.netty.handler.codec.http.HttpResponseStatus.NO_CONTENT;
+import static io.netty.handler.codec.http.HttpResponseStatus.OK;
+import static io.netty.handler.codec.http.HttpResponseStatus.SEE_OTHER;
+import static io.netty.handler.codec.http.HttpResponseStatus.TEMPORARY_REDIRECT;
+import static io.netty.handler.codec.http.HttpVersion.HTTP_1_0;
+import static io.netty.handler.codec.http.HttpVersion.HTTP_1_1;
+import static java.lang.String.valueOf;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.emptyIterable;
+import static org.hamcrest.Matchers.hasItem;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+import static org.hamcrest.Matchers.nullValue;
+import static org.hamcrest.collection.IsIterableContainingInOrder.contains;
+
+public class FullHttpResponseTest {
+    @Test
+    public void encodesToStreamingHttpResponse() {
+        FullHttpResponse<String> response = response()
+                .body("foobar")
+                .build();
+
+        HttpResponse streaming = response.toStreamingHttpResponse(string -> copiedBuffer(string, UTF_8));
+
+        TestSubscriber<ByteBuf> subscriber = TestSubscriber.create(0);
+        subscriber.requestMore(1);
+
+        streaming.body().content().subscribe(subscriber);
+
+        assertThat(subscriber.getOnNextEvents().size(), is(1));
+        ByteBuf buf = subscriber.getOnNextEvents().get(0);
+        assertThat(buf.toString(UTF_8), is("foobar"));
```

Comment:
Also, it should be sufficient to use `streaming.body().decode(...)` to aggregate the content stream for assertions. 

IMO the fact that it is streamed as a single content chunk should be an implementation detail, not part of the method contract (assertion in line 77). That is, we could at any time choose to change the `toStreamingHttpResponse` implementation to split it into several smaller chunks if necessary without affecting the API contract.

Perhaps this could be even clarified in the javadoc comments.